test/conservation_checker_tests.jl

```diff
@@ -14,300 +14,181 @@ using NCDatasets
+    space = TestHelper.create_space(FT)
+
+    # set up model simulations
+    atmos = TestAtmos((; space = space))
+    land = TestOcean((; space = space))
+    ocean = TestLand((; space = space))
+    ice = SurfaceStub((; area_fraction = Fields.ones(space) .* 0.5))
+    model_sims = (; atmos_sim = atmos, land_sim = land, ocean_sim = ocean, ice_sim = ice)
+
+    # conservation checkers
+    cc = (; energy = EnergyConservationCheck(model_sims), water = WaterConservationCheck(model_sims))
+
+    # coupler fields
+    cf = (;
+        F_radiative_TOA = Fields.ones(space),
+        P_net = Fields.zeros(space),
+        P_liq = Fields.zeros(space),
+        P_snow = Fields.zeros(space),
+        F_turb_moisture = Fields.zeros(space),
+    )
+    @. cf.F_radiative_TOA = 200
+    @. cf.P_liq = -100
 
     # init
-    coupler_sim = coupler_sim_from_file(local_file_0, conservation_checks = conservation_checks)
-    rad_source = deepcopy(coupler_sim.fields.F_radiative_TOA)
-    surface_water = deepcopy(coupler_sim.fields.P_net)
-    check_conservation!(coupler_sim, get_slab_energy, get_slab_energy)
-    # 1 day later
-    coupler_sim = coupler_sim_from_file(local_file_end, conservation_checks = conservation_checks)
-    check_conservation!(coupler_sim, get_slab_energy, get_slab_energy)
-    parent(rad_source) .-= parent(deepcopy(coupler_sim.fields.F_radiative_TOA))
-    parent(surface_water) .+= parent(deepcopy(coupler_sim.fields.P_net))
-
-    cc = conservation_checks.energy
-    tot_energy = @. cc.ρe_tot_atmos + cc.ρe_tot_ocean + cc.ρe_tot_land + cc.ρe_tot_seaice + cc.toa_net_source
-    err = ((tot_energy[2] - tot_energy[1]) - sum(rad_source)) / (tot_energy[2])
-    @test abs(err) < 1e-5
-
-    tot_water = conservation_checks.water.ρq_tot_atmos
-    err = ((tot_water[2] - tot_water[1]) - sum(surface_water)) / (tot_water[2])
-    @test abs(err) < 1e-1 # TODO: this could be due to limiters. Need more testing to reduce this error.
-    rm(tmp_dir, recursive = true)
-end
+    cs = Utilities.CoupledSimulation{FT}(
+        nothing, # comms_ctx
+        nothing, # dates
+        space, # boundary_space
+        cf, # fields
+        nothing, # parsed_args
+        cc, # conservation_checks
+        (Int(0), Int(1000)), # tspan
+        Int(200), # t
+        Int(200), # Δt_cpl
+        (;), # surface_masks
+        model_sims, # model_sims
+        (;), # mode
+        (), # diagnostics
+    )
 
-@testset "test check_conservation for land field update" begin
+    # set non-zero radiation and precipitation
+    F_r = cf.F_radiative_TOA
+    P = cf.P_liq
+    Δt = cs.Δt_cpl
+
+    # analytical solution
+    tot_energy_an = sum(F_r .* 3Δt .+ 1e6 .* 1.25)
+    tot_water_an = sum(.-P .* 3Δt .* 0.5 .+ Fields.ones(space))
+
+    # run check_conservation!
+    check_conservation!(cs, runtime_check = true)
+    check_conservation!(cs, runtime_check = true)
+    check_conservation!(cs, runtime_check = true)
```

Comment:
why do we have 3 calls to this in a row?

Reply:
We call them several times to let any potential errors accrue.